vault/audited_headers.go

```diff
@@ -37,7 +41,7 @@ func (a *AuditedHeadersConfig) add(header string, hmac bool) error {
 	a.Lock()
 	defer a.Unlock()
 
-	a.Headers[header] = &auditedHeaderSettings{hmac}
+	a.Headers[strings.ToLower(header)] = &auditedHeaderSettings{hmac}
```

Comment:
Once the binary is rolled, I don't think that this would cause any problems. Otherwise, in weird HA cases (standby not upgraded, leader upgraded, leader steps down and the not upgraded standby assumes leadership), can this lead to multiple entries for headers, like, `x-New-header` and `x-new-header`?